apps/store/themes/store/js/categorization.js

```diff
@@ -17,39 +17,62 @@
  */
 
 var categorizationArray = [];
-var categorization = function() {
+var categorization = function () {
 
-    var updateFilters = function (url){
-        var queryArray = url.split("q=")[1].split(",");
+    /**
+     * This method returns the query parameter by name
+     * @param name          name of the query parameter
+     * @param url           url
+     * @returns {string}    query parameter string
+     */
+    var getParameterByName = function (name, url) {
+        if (!url) {
+            url = window.location.href;
+        }
+        var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)");
+        var results = regex.exec(url);
+        if (results && results[2]) {
+            return decodeURIComponent(results[2].replace(/\+/g, " "));
+        }
+
+        return '';
+    };
+
+    var updateFilters = function (url) {
+        var queryParam = getParameterByName('q', url);
+        var queryArray = [];
+        if (queryParam) {
+            queryArray = queryParam.split(",");
+        }
         var queryObjArray = [];
-        if(queryArray.length > 0){
-            for(var index in queryArray){
+        if (queryArray.length > 0) {
+            for (var index in queryArray) {
                 var obj = queryArray[index];
                 var queryObj = {};
                 var queryObjValue = [];
                 queryObj.queryKey = obj.split(":")[0].split('\"').join('');
-                if(obj.indexOf("(") > -1){
+                if (obj.indexOf("(") > -1) {
                     var newObj = obj.split(":")[1].split('\"').join('').replace("(", "").replace(")", "");
                     queryObjValue = newObj.split("OR");
                 } else {
-                    if(obj.indexOf(":") > -1){
+                    if (obj.indexOf(":") > -1) {
                         queryObjValue.push(obj.split(":")[1].split('\"').join(''));
                     }
                 }
                 queryObj.queryValue = queryObjValue;
                 queryObjArray.push(queryObj);
             }
 
-            for(var i in queryObjArray){
-                $('.categorization-checkbox:checkbox').each(function() {
+            for (var i in queryObjArray) {
```

Comment:
@mushthaq33 this code block is replaced by 6fe1ed65758f492c9d025ff984bd593e35c04530